src/client/helpers/entity.js

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2016  Daniel Hsing
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+/* eslint-disable react/display-name */
+const React = require('react');
+const FontAwesome = require('react-fontawesome');
+const Button = require('react-bootstrap').Button;
+const Table = require('react-bootstrap').Table;
+const formatDate = require('./utils').formatDate;
+const _ = require('lodash');
+
+function extractAttribute(attr, path) {
+	'use strict';
+	if (attr) {
+		if (path) {
+			return _.get(attr, path, '?');
+		}
+		return attr;
+	}
+	return '?';
+}
+
+function getLanguageAttribute(entity) {
+	'use strict';
+	const languages = (entity.languageSet && entity.languageSet.languages) ?
+		entity.languageSet.languages.map(
+			(language) => language.name
+		).join(', ') : '?';
+	return {title: 'Languages', data: languages};
+}
+
+function getTypeAttribute(entityType) {
+	'use strict';
+	return {title: 'Type', data: extractAttribute(entityType, 'label')};
+}
+
+function getDateAttributes(entity) {
+	'use strict';
+	const attributes = [{
+		title: 'Begin Date',
+		data: extractAttribute(formatDate(new Date(entity.beginDate)))
```

Comment:
Just noticed, if `entity.beginDate` is null, the formatted date will be 1970-01-01 - need to add a condition to prevent this.